src/contexts/auth-context.js

```diff
@@ -0,0 +1,108 @@
+import axios from "axios";
+import { useNavigate, useLocation } from "react-router-dom";
+import React, { useState, useContext, createContext } from "react";
+
+const authContext = createContext({ isUserLoggedIn: false });
+
+const AuthProvider = ({ children }) => {
+
+    const [isUserLoggedIn, setIsUserLoggedIn] = useState(false)
+    const [formData, setFormData] = useState({ userEmail: "", userPwd: "" });
+    const [encodedData, setEncodedData] = useState("");
+    const [signUpErrMsg, setSignUpErrMsg] = useState();
+    const [logInErrMsg, setLogInErrMsg] = useState("");
+    const navigate = useNavigate();
+    const location = useLocation();
+
+    const signUpHandler = async (e) => {
+        e.preventDefault();
+        if (formData.userEmail !== "" && formData.userPwd !== "") {
+            if (/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(formData.userEmail)) {
```

Comment:
I think you can use a `pattern` attribute in the `input` tag to validate the input

Reply:
okk didn't know about about this attribute, thanks for suggesting